### PR TITLE
Issue #1 invalid username - state changes discussed

### PR DESCRIPTION
**Pull request by buglePhotographer**

### CODE REVIEW

File: web/src/App.js [hunk 1]
```diff
@@ -72,9 +89,19 @@ class App extends Component {
...
         </HeightMotion>
-        <HeightMotion show={this.isLoading} height={30}>
+        <HeightMotion
+          show={state.username && state.tweets.length === 0 && !state.error}
+          height={30}
+        >
           <StatusBar username={state.username} />
         </HeightMotion>
+        <HeightMotion
+          show={state.tweets.length === 0 && state.error}
```
Comment (pomber) [inline web/src/App.js]: This could probably be just `show={state.error}`

File: web/src/App.js [hunk 2]
```diff
@@ -63,7 +77,10 @@ class App extends Component {
         <HeightMotion show={!state.username} height={40} className="landing">
           Enter any twitter username to auto-generate fake tweets:
         </HeightMotion>
-        <FormMotion loading={this.isLoading} onChange={this.handleLoad} />
+        <FormMotion
+          loading={state.username && state.tweets.length === 0 && !state.error}
```
Comment (pomber) [inline web/src/App.js]: Why not use `this.isLoading` here? We could add `&& !state.error` there

File: web/src/App.js [hunk 3]
```diff
@@ -72,9 +89,19 @@ class App extends Component {
             Hexacta Innovation Labs
           </a>
         </HeightMotion>
-        <HeightMotion show={this.isLoading} height={30}>
+        <HeightMotion
+          show={state.username && state.tweets.length === 0 && !state.error}
```
Comment (pomber) [inline web/src/App.js]: Same here

File: web/src/App.js [hunk 4]
```diff
@@ -33,19 +30,36 @@ class App extends Component {
...
-      tweets: []
+      tweets: [],
+      error: false
     });
-    service.getInfo(username).then(this.load);
+    service.getInfo(username).then(this.load).catch(this.loadError);
   };
 
   load = info => {
+    if (!info && !info.model) {
```
Comment (pomber) [inline web/src/App.js]: When is this true? Aren't info and info.model always set when `load` is called?

**Review by pomber — CHANGES_REQUESTED** [review]
Almost there, just a few changes left.   
FYI: instead of opening a new PR for each change you can commit to the same branch and changes will be added to this PR.

File: web/src/App.js (hunk 1 above)
Comment (buglePhotographer) [reply web/src/App.js]: Yeah.

File: web/src/App.js (hunk 3 above)
Comment (buglePhotographer) [reply web/src/App.js]: Yeah ok.

File: web/src/App.js (hunk 2 above)
Comment (buglePhotographer) [reply web/src/App.js]: Yeah ok.

File: web/src/App.js (hunk 4 above)
Comment (buglePhotographer) [reply web/src/App.js]: The model does not come filled when there's an error, like the one we are handling here. Model comes undefined, so no user data retrieved.

**Review by pomber — APPROVED** [review]

**Comment by pomber** [comment]
Merged. Thanks a lot!

**Comment by buglePhotographer** [comment]
Thank you.

**Pull request merged**